File: problems/925-long-pressed-name/solution.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {
        int i = 0, j = 0;
        while (i < name.length() && j < typed.length()) {
            if (name.charAt(i) == typed.charAt(j)) {
                i++;
                j++;
            } else {
                j++;
            }
        }
        if (i == name.length()) {
            while (j < typed.length()) {
                if (typed.charAt(j) != typed.charAt(j-1)) return false;
                j++;
            }
            return true;
        } else {
            return false;
        }
    }
}

File: problems/925-long-pressed-name/solution.c
bool isLongPressedName(char* name, char* typed) {
    int i = 0, j = 0;
    while (i < strlen(name) && j < strlen(typed)) {
        if (name[i] == typed[j]) {
            int count1 = 0, count2 = 0;
            char c = name[i];
            while (i < strlen(name) && name[i] == c) {
                count1++;
                i++;
            }
            while (j < strlen(typed) && typed[j] == c) {
                count2++;
                j++;
            }
            if (count2 < count1) return false;
        } else {
            return false;
        }
    }
    return i == strlen(name);
}